allennlp/training/metrics/categorical_accuracy.py

```diff
@@ -44,18 +52,29 @@ def __call__(self,
             raise ConfigurationError("A gold label passed to Categorical Accuracy contains an id >= {}, "
                                      "the number of classes.".format(num_classes))
 
-        # Top K indexes of the predictions (or fewer, if there aren't K of them).
-        # Special case topk == 1, because it's common and .max() is much faster than .topk().
-        if self._top_k == 1:
-            top_k = predictions.max(-1)[1].unsqueeze(-1)
-        else:
-            top_k = predictions.topk(min(self._top_k, predictions.shape[-1]), -1)[1]
+        predictions = predictions.view((-1, num_classes))
+        gold_labels = gold_labels.view(-1).long()
+        if not self._tie_break:
+            # Top K indexes of the predictions (or fewer, if there aren't K of them).
+            # Special case topk == 1, because it's common and .max() is much faster than .topk().
+            if self._top_k == 1:
+                top_k = predictions.max(-1)[1].unsqueeze(-1)
+            else:
+                top_k = predictions.topk(min(self._top_k, predictions.shape[-1]), -1)[1]
 
-        # This is of shape (batch_size, ..., top_k).
-        correct = top_k.eq(gold_labels.long().unsqueeze(-1)).float()
+            # This is of shape (batch_size, ..., top_k).
+            correct = top_k.eq(gold_labels.unsqueeze(-1)).float()
+        else:
+            # prediction is correct if gold label falls on any of the max scores. distribute score by tie_counts
+            max_predictions = predictions.max(-1)[0]
+            max_predictions_mask = predictions.eq(max_predictions.unsqueeze(-1).expand_as(predictions))
```

Comment:
Why do you need the `expand_as` here? I think this is already the same shape as `predictions`.

Reply:
If predictions is `batch_size X num_classes`, then max_predictions is `batch_size` because it maxed out on last dim. So I unsqueeze it to make `batch_size X 1` and then `expand_as` so as to copy it across last dimension (and so `batch_size X num_classes`).

Reply:
Oh sorry - I didn't see that this expand was inside the `.eq`. I still don't think you need it, because just be expanding, the `eq` will broadcast along the dimensions.

Reply:
Ohh right! Done and works perfect!